mobx/lib/src/api/reaction.dart

```diff
@@ -24,11 +24,11 @@ import 'package:mobx/src/core.dart';
 /// x.value = 30; // Will not cause autorun() to re-trigger as it's disposed.
 /// ```
 
-ReactionDisposer autorun(Function(Reaction) fn,
+ReactionDisposer autorun(Function(ReactionImpl) fn,
```

Comment:
I am not in favor of exposing `ReactionImpl` in the public signatures. Ideally the `Reaction` (the interface) is what we are exposing, whose implementation happens to be `ReactionImpl`. I think we should keep the interface here in the signature, like it was before:

`ReactionDisposer autorun(Function(Reaction) fn)`

Reply:
Same goes for all the other signatures. Since `ReactionImpl` implements **Reaction**, we should code to the interface rather than the implementation.